03-intro-telemetry-data.qmd

```diff
@@ -392,87 +491,72 @@ ggplot(sbou_dur, aes(diff)) +
 
 
-To create a standardized fix per day, lets take the first fix per day. This could be based on a number of factors, depending on our research question we want to ask 
-
+To create a standardized fix per day, lets take the first fix per day. This could be based on a number of factors, depending on our research question we want to ask. We can use Julian date to filter all fixes. 
 
 ```{r}
-
 # we can subset base on a Julian date
-
-sbou_dur <- sbou_dur %>%
+sbou_dur <- sbou_dur |> 
   mutate(
-    date2=as.Date(date_time, format = '%Y-%m-%d'),
-    jdate=julian(date2)
+    date2=as.Date(date_time, format = '%Y-%m-%d'), # convert to date type
+    jdate=julian(date2)               # create new column and calculate Julian date 
   )
 
-# Check the multiple counts of anaimals per day 
-counts.per.day.jdate <- sbou_dur %>% 
-  group_by(animal.id, jdate, year) %>% 
-  summarise(total = n(), unique = unique(jdate)) %>% 
-  group_by(animal.id, total, year) %>% 
+
+# Check the multiple counts of animals per day 
+counts.per.day.jdate <- sbou_dur |> 
+  group_by(animal.id, jdate, year) |> 
+  summarise(total = n(), unique = unique(jdate)) |> 
+  group_by(animal.id, total, year) |> 
   summarise(total.j = n()) 
 
 
+# lets plot this data to make more sense of what is happening
 ggplot(counts.per.day.jdate, aes(x = total, y = total.j)) + 
   geom_point() +
   geom_vline(xintercept = 1, color = "red")+
   facet_wrap(.~animal.id + year)
 
 
-sbou_sub <- sbou_dur %>%
-    group_by(animal.id, jdate) %>%
+# lets select the first fix per day 
+sbou_sub <- sbou_dur |> 
+    group_by(animal.id, jdate) |> 
     slice_sample( n = 1) 
 
 
-# repeat this
-
-# Check the multiple counts of anaimals per day 
-counts.per.day.jdate <- sbou_sub %>% group_by(animal.id, jdate, year) %>% 
-  summarise(total = n(), unique = unique(jdate)) %>% 
-  group_by(animal.id, total, year) %>% 
-  summarise(total.j = n()) 
-
-
-ggplot(counts.per.day.jdate, aes(x = total, y = total.j)) + 
-  geom_point() +
-  geom_vline(xintercept = 1, color = "red") +
-  facet_wrap(.~animal.id + year)
-
-
-length(sbou_sub$location.long)
 ```
 
-```{r}
+We can write out the standardized fix rate file for future use in analysis.
+
+```{r write-standard-subset-csv}
 #| eval: !expr '!file.exists(file.path("clean_data", "scott_herd_subset.csv")) || isTRUE(params$write)'
 # lets write this out as .csv and .Gpkg
 
-write_csv(sbou_sub, file.path("clean_data", "scott_herd_subset.csv"))
-```
+write_csv(sbou_sub, "clean_data/scott_herd_subset.csv")
 
+```
 
-convert to a sf object 
+Alternatively we can convert to a sf object and export. Note this is required as we previously remove the spatial components to undertake the summaries above.
 
-```{r}
-sbou_sub_sf <- st_as_sf(sbou_sub, coords = c("location.long", "location.lat"), 
+```{r write-standard-subset-gpk}
+sbou_sub_sf <- read_sf(sbou_sub, coords = c("location.long", "location.lat"), 
```

Comment:
I think `st_as_sf()` was correct here? `sbou_sub` is a regular data frame right?
```suggestion
sbou_sub_sf <- st_as_sf(sbou_sub, coords = c("location.long", "location.lat"), 
```

Reply:
good catch - thanks i think i just was doing it on auto pilot